openassessment/xblock/staff_area_mixin.py

```diff
@@ -467,60 +462,57 @@ def clear_student_state(self, user_id, course_id, item_id, requesting_user_id):
             'item_id': item_id,
             'item_type': 'openassessment',
         }
-        # There *should* only be one submission, but the logic is easy to extend for multiples so we may as well do it
         submissions = submission_api.get_submissions(student_item)
-        for sub in submissions:
-            # Remove the submission from grading pools
-            self._cancel_workflow(sub['uuid'], "Student state cleared", requesting_user_id=requesting_user_id)
-
-            # Tell the submissions API to orphan the submission to prevent it from being accessed
-            submission_api.reset_score(
-                user_id,
-                course_id,
-                item_id,
-                clear_state=True
-            )
 
-    def clear_team_state(self, user_id, course_id, item_id, requesting_user_id):
+        if self.is_team_assignment():
+            self.clear_team_state(user_id, course_id, item_id, requesting_user_id, submissions)
+        else:
+            # There *should* only be one submission, but the logic is easy to extend for multiples so we may as well
+            for sub in submissions:
+                # Remove the submission from grading pools
+                self._cancel_workflow(sub['uuid'], "Student state cleared", requesting_user_id=requesting_user_id)
+
+                # Tell the submissions API to orphan the submission to prevent it from being accessed
+                submission_api.reset_score(
+                    user_id,
+                    course_id,
+                    item_id,
+                    clear_state=True
+                )
+
+    def clear_team_state(self, user_id, course_id, item_id, requesting_user_id, submissions):
         """
         This is called from clear_student_state (which is called from the LMS runtime) when the xblock is a team
         assignment, to clear student state for an entire team for a given problem. It will cancel the workflow
         to remove it from the grading pools, and pass through to the submissions team API to orphan the team
         submission and individual submissions so that the team can create a new submission.
         """
-        error_msg_base = 'Attempted to clear team state for anonymous user {} '.format(user_id)
-        try:
-            user_team = self.get_team_for_anonymous_user(user_id)
-        except ObjectDoesNotExist:
-            warning_msg = error_msg_base + 'but was unable to resolve to a real user'
-            logger.warning(warning_msg)
-            return
+        student_item_string = "course {} item {} user {}".format(course_id, item_id, user_id)
 
-        if user_team is None:
-            warning_msg = error_msg_base + 'but they are not on a team for course {} item {}.'.format(
-                course_id, item_id
-            )
-            logger.warning(warning_msg)
+        if not submissions:
```

Comment:
Trivial nit: we are mixing string.format and %s for string formatting.

Reply:
Yeah, but at the top i'm building a string and then using that built string as a %s format parameter itself, idk i just wanted to try to save myself some lines